bin/varnishd/hash/hash_critbit.c

```diff
@@ -364,7 +366,7 @@ hcb_deref(struct objhead *oh)
 #ifdef PHK
 	fprintf(stderr, "hcb_defef %d %d <%s>\n", __LINE__, r, oh->hash);
 #endif
-	return (1);
+	return (r);
```

Comment:
HSH_DeleteObjHead(wrk, oh); missing for r==0 ?

Reply:
Don't think so. Critbit puts it on the cooling list when r==0, and it will be picked up and free'd later. The cooling list is what enables the critbit code to use the objhead's own mutex for protecting it's refcount.

Reply:
I see it now, thank you. The reason for my confusion is that in the old HSH_DerefObjHead there was a HSH_DeleteObjHead for r==0, but that never happened, and which you now removed. So I thought it had to go here, but I missed the fixed return(1) here. Thank you, I agree now